roofit/roofitcore/src/RooRealSumPdf.cxx

```diff
@@ -159,7 +151,7 @@ RooRealSumPdf::RooRealSumPdf(const char *name, const char *title,
   if (!(inFuncList.getSize()==inCoefList.getSize()+1 || inFuncList.getSize()==inCoefList.getSize())) {
     coutE(InputArguments) << "RooRealSumPdf::RooRealSumPdf(" << GetName() 
 			  << ") number of pdfs and coefficients inconsistent, must have Nfunc=Ncoef or Nfunc=Ncoef+1" << endl ;
-    assert(0) ;
+    throw std::invalid_argument("RooRealSumPdf: Number of PDFs and coefficients is inconsistent.");
```

Comment:
this is only NFC is you expect no-one ever will be executing this within a try/catch (maybe much further up in the stack). is execution actually safely resumable (with no leaks, for instance) if the program reaches this state (or the others below)? Otherwise I'd suggest using a printout + `std::abort` rather than an exception.